extensions/amp-social-share/1.0/social-share.js

```diff
@@ -0,0 +1,210 @@
+    const windowFeatures = 'resizable,scrollbars,width=640,height=480';
+    if (protocol === 'navigator-share') {
+      if (window && window.navigator && window.navigator.share) {
+        const dataStr = finalEndpoint.substr(finalEndpoint.indexOf('?'));
+        const data = parseQueryString(dataStr);
+        window.navigator.share(data).catch((e) => {
+          throwWarning(`${e.message}. ${name}`);
+        });
+      } else {
+        throwWarning(
+          `Could not complete system share.  Navigator unavailable. ${name}`
+        );
+      }
+    } else if (protocol === 'sms') {
+      openWindowDialog(
+        window,
+        finalEndpoint.replace('?', '?&'),
+        '_blank',
+        windowFeatures
+      );
+    } else {
+      openWindowDialog(window, finalEndpoint, '_blank', windowFeatures);
+    }
+  }
+
+  /**
+   * @private
+   * @param {!Event} event
+   */
+  function handleKeyPress(event) {
+    const {key} = event;
+    if (key == Keys.SPACE || key == Keys.ENTER) {
+      event.preventDefault();
+      handleActivation();
+    }
+  }
+
+  const type = props.type.toUpperCase();
+  const baseStyle = CSS.BASE_STYLE;
+  const backgroundStyle = CSS[type] || CSS.DEFAULT;
+  const size = {
+    width: checkedWidth,
+    height: checkedHeight,
+  };
+  return (
+    <div
+      role="button"
+      tabindex={props['tabIndex'] || '0'}
+      onKeyDown={handleKeyPress}
+      onClick={handleActivation}
+      style={{...size, ...props['style']}}
+      {...props}
+    >
+      <SocialShareIcon
+        style={{...backgroundStyle, ...baseStyle, ...size}}
+        type={type}
+      />
+    </div>
+  );
+}
+
+/**
+ * @param {!JsonObject} props
+ * @param {?string} name
+ * @return {!JsonObject}
+ */
+function checkProps(props, name) {
+  const {type, shareEndpoint, params, bindings, width, height} = props;
+
+  // Verify type is provided
+  if (type === undefined) {
+    throw new Error(`The type attribute is required. ${name}`);
+  }
+
+  // bindings and params props must be objects
+  if (params && !isObject(params)) {
+    throw new Error(`The params property should be an object. ${name}`);
+  }
+  if (bindings && !isObject(bindings)) {
+    throw new Error(`The bindings property should be an object. ${name}`);
+  }
+
+  // User must provide shareEndpoint if they choose a type that is not
+  // pre-configured
+  const typeConfig = getSocialConfig(type) || dict();
+  const baseEndpoint = typeConfig['shareEndpoint'] || shareEndpoint;
+  if (baseEndpoint === undefined) {
+    throw new Error(
+      `A shareEndpoint is required if not using a pre-configured type. ${name}`
+    );
+  }
+
+  // Verify width and height are valid integers
+  // Silently assigns default for undefined, null
+  // Throws Warning for booleans, strings, 0, negative numbers
+  // No errors when positive integer or equivalent string
+  let checkedWidth =
+    width === null || width === undefined ? DEFAULT_WIDTH : width;
```

Comment:
`x === null || x === undefined` is `x == null`. But also we should be able to use `??` now. E.g. this expression can be done as `width ?? DEFAULT_WIDTH`.

Reply:
Updated to `??` and removed the if statement underneath.